src/map/skill.cpp

```diff
@@ -8838,6 +8838,9 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 			if(md->db->skill[md->skill_idx].val[4] && tsce)
 				status_change_end(bl, type, INVALID_TIMER);
 
+			if (battle_config.npc_emotion_behavior && skill_id == NPC_EMOTION && md->db->skill[md->skill_idx].val[1] && md->db->skill[md->skill_idx].val[1] == 0x81)
```

Comment:
I don't get why you do this

Reply:
I've created this condition because I wanted to check exactly when NPC EMOTION change the mode into 0x81 (which means MD_CANMOVE plus MD_CANATTACK mode). This was the mode I've observed that turns the mob into a passive state on mob_skill_db.

Reply:
I've updated the code following the other changes you requested on commit a441fb2.

Reply:
dont put magic numbers then.
and rename the battle config to "disable aggressiveness toggling" or something

Reply:
The magic number is already taken off (check commit a441fb2). About the name, I've choosen "npc_emotion_behavior" initially because it's directly related to the NPC EMOTION skill. And it disables passiveness, not the opposite. What about the name "npc_emotion_disable_passiveness"?